pkg/cli/klothomain.go

```diff
@@ -180,9 +180,21 @@ func (km KlothoMain) run(cmd *cobra.Command, args []string) (err error) {
 		zap.S().Warnf("failed to create .klotho directory: %v", err)
 	}
 
+	analyticsClientProperties := map[string]interface{}{
+		"version": km.Version,
+		"strict":  cfg.strict,
+		"edition": km.DefaultUpdateStream,
+	}
+
 	// Set up user if login is specified
 	if cfg.login {
-		err := auth.Login()
+		err := auth.Login(func(err error) {
+			// We don't have the analytics client set up to the logger yet, so the warn message won't send any
```

Comment:
Is there a reason we couldn't set up the analytics client earlier? Doesn't seem like it needs anything setup in this block.

Reply:
`analytics.NewClient` looks for login creds (either authenticated or local), and errors out if they're not there — so you wouldn't be able to log in until you've logged in, which means our only user would be Yossarian.

This flow does feel pretty janky though, I agree. I think a nicer thing would be to first construct the client without login info, then do any possible login, then attach the login info. So:

1. Does that revised workflow make sense, and if so
2. Should I do it in this PR (which is into the `auth` branch, which will eventually need full review to land into `main`) or as a separate, followup cleanup task?

Reply:
> I think a nicer thing would be to first construct the client without login info, then do any possible login, then attach the login info.

👍 
This branch or a different PR doesn't matter to me.

Reply:
Filed as #180